File: MergeTwoSortedLists/MergeSortedListsAns_Easy.java

// useful imports; you may not need these depending on your approach

public class MergeSortedListsAns_Easy {
    int[] arr1;
    int[] arr2;

    // constructor
    public MergeSortedListsAns_Easy(int[] arr1, int[] arr2) {
        this.arr1 = arr1;
        this.arr2 = arr2;
    }

    // solution method -- edit this
    public int[] mergeMethod() {
        int[] outputList = new int[this.arr1.length + this.arr2.length];

        int ptr1 = 0;
        int ptr2 = 0;

        // for all elements that will be added into the new merged array
        for (int i = 0; i < outputList.length; i++) {
            // if both indices are in range
            if (ptr1 < this.arr1.length && ptr2 < this.arr2.length) {
                if (this.arr1[ptr1] <= this.arr2[ptr2]) {
                    outputList[i] = this.arr1[ptr1];
                    ptr1++;
                } else {
                    outputList[i] = this.arr2[ptr2];
                    ptr2++;
                }
            }
            // if arr1 is in range but arr2 is not
            else if (ptr1 < this.arr1.length) {
                outputList[i] = this.arr1[ptr1];
                ptr1++;
            }
            // if arr2 is in range but arr1 is not
            else if (ptr2 < this.arr2.length) {
                outputList[i] = this.arr2[ptr2];
                ptr2++;
            } else {
                break;
            }
        }
        return outputList;
    }

    public static void main(String[] args) {
        // instantiate class and call solver method here
        int[] a = { 1, 4, 6 };
        int[] b = { 2, 5, 12, 54 };
        MergeSortedListsAns_Easy test = new MergeSortedListsAns_Easy(a, b);
        int[] mergedArr = test.mergeMethod();

        for (int i = 0; i < mergedArr.length; i++) {
            System.out.print(mergedArr[i] + " ");
        }
        System.out.println();
    }
}

File: MergeTwoSortedLists/MergeSortedListsAns_Easy.py
# prompt - https://leetcode.com/problems/merge-two-sorted-lists/

# obj - merge two sorted lists into one sorted list

class Solution:
    # constructor
    def __init__(self, arr1, arr2) -> None:
        self.arr1 = arr1
        self.arr2 = arr2

    # merge method -- simple
    def merge(self):
        # instantiate empty arr to hold sorted values
        mergedArr = []

        # series of pointers
        ptrArr1 = 0
        ptrArr2 = 0

        # set infinite loop
        while True:
            # if pointer 1 out of bounds, add the rest of arr2 and break out
            if ptrArr1 == len(self.arr1):
                for i in range(ptrArr2, len(self.arr2)):
                    mergedArr.append(self.arr2[i])
                break

            # if pointer 2 out of bounds, add the rest of arr1 and break out
            elif ptrArr2 == len(self.arr2):
                for i in range(ptrArr1, len(self.arr1)):
                    mergedArr.append(self.arr1[i])
                break

            # if both pointers in bounds, compare values, add min of these two to mergedArr, and incrememnt as needed
            if self.arr1[ptrArr1] <= self.arr2[ptrArr2]:
                mergedArr.append(self.arr1[ptrArr1])
                ptrArr1 += 1
            elif self.arr1[ptrArr1] > self.arr2[ptrArr2]:
                mergedArr.append(self.arr2[ptrArr2])
                ptrArr2 += 1

            # probably not needed, but does not hurt :-)
            else:
                break
        # return sorted array
        return mergedArr


b = [1, 4, 6]
c = [2, 5, 12, 54]
a = Solution(b, c).merge()
print(a)
